tasks/init.rb

```diff
@@ -0,0 +1,137 @@
+  # the value of :body_format. If Content-Type is specified in :headers, that
+  # value will take priority.
+  #
+  def make_headers(headers, body_format, user_agent)
+    base = case body_format
+           when 'form-urlencoded'
+            { 'Content-Type' => 'application/x-www-form-urlencoded' }
+           when 'json'
+            { 'Content-Type' => 'application/json' }
+           when 'raw'
+            { 'Content-Type' => 'text/plain' }
+           end
+
+    base['User-Agent'] = user_agent
+
+    base.merge!(headers || {})
+  end
+
+  # Formats the body for the request. If the Content-Type is application/json,
+  # this will convert the body to JSON. Otherwise, the body will be stringified.
+  #
+  def make_body(body, body_format)
+    return unless body
+
+    case body_format
+    when 'form-urlencoded'
+      debug 'URL encoding request body'
+      URI.encode_www_form(body)
+    when 'json'
+      debug 'Serializing request body as JSON'
+      json?(body) ? body : body.to_json
+    else
+      debug 'Stringifying request body'
+      body.to_s
+    end
+  rescue StandardError
+    raise TaskHelper::Error.new(
+      'Unable to format request body',
+      'http_request/body-format-error',
+      'details' => { 'debug' => debug_statements, 'format' => body_format }
+    )
+  end
+
+  # Returns true if the data is already formatted as JSON, false otherwise.
+  #
+  def json?(data)
+    JSON.parse(data)
+    true
+  rescue StandardError
+    false
+  end
+
+  # Makes the actual request.
+  def request(method, uri, headers, body, opts = {})
+    # Create the client
+    client = Net::HTTP.new(uri.host, uri.port)
+  
+    # Use SSL if requesting a secure connection
+    if uri.scheme == 'https'
+      client.use_ssl     = true
+      client.verify_mode = OpenSSL::SSL::VERIFY_PEER
+      client.ca_file     = opts[:cacert] if opts[:cacert]
+      client.cert        = OpenSSL::X509::Certificate.new(File.read(opts[:cert])) if opts[:cert]
+      client.key         = OpenSSL::PKey::RSA.new(opts[:key]) if opts[:key]
+    end
+  
+    # Build the request
+    request = Net::HTTP.const_get(method).new(uri.request_uri, headers)
+  
+    # Build the query if there's data to send
+    request.body = body if body
+  
+    # Send the request
+    client.request(request)
+  rescue StandardError => e
+    raise TaskHelper::Error.new(
+      "Failed to connect to #{uri}: #{e.message}",
+      'http_request/connect-error'
+    )
+  end
+
+  # Handles the response. First checks the response code and determines whether
+  # the task will succeed or fail depending on the list of :success_codes. Next,
+  # it will check the headers to determine if the response body should be parsed
+  # as json.
+  #
+  def handle_response(response, success_codes)
+    json = response['Content-Type'] == 'application/json'
+
+    details = {
+      body:        json ? JSON.parse(response.body) : response.body,
```

Comment:
A common pain point in dealing with JSON APIs is hitting the wrong route and getting a non-json response. I think it would be better if there was a way to make that an error rather than a completely unexpected structure for the response.